packages/core/src/middleware/koa-oidc-interaction.ts

```diff
@@ -0,0 +1,22 @@
+import { MiddlewareType } from 'koa';
+import { IRouterParamContext } from 'koa-router';
+import { Provider } from 'oidc-provider';
+
+export type Interaction = Awaited<ReturnType<Provider['interactionDetails']>>;
+
+export type WithOidcInteractionContext<ContextT extends IRouterParamContext = IRouterParamContext> =
+  ContextT & {
+    interaction: Interaction;
+  };
+
+export default function koaOidcInteraction<
+  StateT,
+  ContextT extends IRouterParamContext,
+  ResponseBodyT
+>(provider: Provider): MiddlewareType<StateT, WithOidcInteractionContext<ContextT>, ResponseBodyT> {
+  return async (ctx, next) => {
+    ctx.interaction = await provider.interactionDetails(ctx.req, ctx.res);
```

Comment:
should we load all the interaction info into the context?

Reply:
just extract the needed part from `Interaction`?

See `packages/core/src/routes/session.ts`.
We need to extract different info from `Interaction` in different situations.

Maybe it's easier to understand and keep using `Interaction` directly?
Or have to modify `koaOidcInteraction` when we need more info from `Interaction`.

![image](https://user-images.githubusercontent.com/10594507/167342393-59ef8977-d37e-4679-96a2-e5852911c025.png)

![image](https://user-images.githubusercontent.com/10594507/167342424-a61f83f1-2fd6-43b2-aaeb-1ac1b9e3c9aa.png)

Reply:
See Slack thread [discussion](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1652079610020569) for more.